tensorflow/core/kernels/generate_box_proposals_op.cu.cc

```diff
@@ -0,0 +1,568 @@
+    const auto height = scores.dim_size(1);
+    const auto width = scores.dim_size(2);
+    const auto box_dim = anchors.dim_size(2) / num_anchors;
+    OP_REQUIRES(context, box_dim == 4,
+                errors::OutOfRange("Box dimensions need to be 4"));
+    // TODO(skama): make sure that inputs are ok.
+    const int image_stride = height * width;
+    const int conv_layer_nboxes =
+        image_stride *
+        num_anchors;  // total number of boxes when decoded on anchors.
+    // The following calls to CUB primitives do nothing
+    // (because the first arg is nullptr)
+    // except setting cub_*_temp_storage_bytes
+    float nms_threshold;
+    int pre_nms_topn;
+    float min_size;
+    OP_REQUIRES_OK(context, GetScalarValue(context, 4, &nms_threshold));
+    if (nms_threshold < 0 || nms_threshold > 1.0) {
+      context->SetStatus(errors::InvalidArgument(
+          "nms_threshold should be between 0 and 1. Got ", nms_threshold));
+      return;
+    }
+    OP_REQUIRES_OK(context, GetScalarValue(context, 5, &pre_nms_topn));
+    if (pre_nms_topn <= 0) {
+      context->SetStatus(errors::InvalidArgument(
+          "pre_nms_topn should be greater than 0", pre_nms_topn));
+      return;
+    }
+    OP_REQUIRES_OK(context, GetScalarValue(context, 6, &min_size));
+    auto cuda_stream = GetGpuStream(context);
+    size_t cub_sort_temp_storage_bytes = 0;
+    float* flt_ptr = nullptr;
+    int* int_ptr = nullptr;
+    cudaError_t cuda_ret = cub::DeviceSegmentedRadixSort::SortPairsDescending(
+        nullptr, cub_sort_temp_storage_bytes, flt_ptr, flt_ptr, int_ptr,
+        int_ptr, num_images * conv_layer_nboxes, num_images, int_ptr, int_ptr,
+        0, 8 * sizeof(float),  // sort all bits
+        cuda_stream);
+    TF_OP_REQUIRES_CUDA_SUCCESS(context, cuda_ret);
+    // get the size of select temp buffer
+    size_t cub_select_temp_storage_bytes = 0;
+    char* char_ptr = nullptr;
+    float4* f4_ptr = nullptr;
+    TF_OP_REQUIRES_CUDA_SUCCESS(
+        context, cub::DeviceSelect::Flagged(
+                     nullptr, cub_select_temp_storage_bytes, f4_ptr, char_ptr,
+                     f4_ptr, int_ptr, image_stride * num_anchors, cuda_stream));
+    Tensor d_conv_layer_indexes;  // box indices on device
+    Tensor d_image_offset;        // starting offsets boxes for each image
+    Tensor d_cub_temp_buffer;     // buffer for cub sorting
+    Tensor d_sorted_conv_layer_indexes;  // output of cub sorting, indices of
+                                         // the sorted boxes
+    Tensor dev_sorted_scores;            // sorted scores, cub output
+    Tensor dev_boxes;                    // boxes on device
+    Tensor dev_boxes_keep_flags;  // bitmask for keeping the boxes or rejecting
+                                  // from output
+    const int nboxes_to_generate = std::min(conv_layer_nboxes, pre_nms_topn);
+    size_t cub_temp_storage_bytes=max(cub_sort_temp_storage_bytes,cub_select_temp_storage_bytes);
```

Comment:
`third_party/tensorflow/core/kernels/generate_box_proposals_op.cu.cc:398:9: error: no matching function for call to 'max'
        max(cub_sort_temp_storage_bytes, cub_select_temp_storage_bytes);`

can you please check above error